### PR TITLE
NDRS-641: PutDeploy RPC endpoint should return an error if the Deploy is invalid

### PR DESCRIPTION
**Pull request by darthsiroftardis**
When a client submits a deploy, a responder object is now passed to the `deploy_acceptor` such that the responder now returns an error code in the case of an invalid deploy

Sample Output:
Client:
`thread 'main' panicked at 'unable to put deploy ResponseIsError(Error { code: 32008, message: "invalid deploy", data: None })', client/src/deploy/transfer.rs:202:31`

Node Log:
`{"timestamp":"Dec 16 11:21:51.916","level":"INFO","fields":{"message":"the deploy submitted by the client was invalid","deploy_hash":"deploy-hash(49bf..6276)"},"target":"casper_node::components::rpc_server::rpcs::account"}`

### CODE REVIEW

**Review by dwerner — APPROVED** [review]
lgtm! Probably best to wait for another approval before merging.

File: node/src/components/deploy_acceptor.rs
Comment (EdHastingsCasperLabs) [inline node/src/components/deploy_acceptor.rs:134]: Add a comment to explain why you are doing this please, for the benefit of the next person to work in this file...which might even be future you!

File: node/src/effect.rs
Comment (EdHastingsCasperLabs) [inline node/src/effect.rs:195]: why are you reducing this to warn ?

File: node/src/components/deploy_acceptor.rs
Comment (EdHastingsCasperLabs) [inline node/src/components/deploy_acceptor.rs:83]: include in doc comment when it is appropriate to pass None (from peer) vs Some (from rpc).

File: node/src/effect.rs
Comment (darthsiroftardis) [reply node/src/effect.rs:195]: That was an accident, a result of some debugging, I shall correct it.

File: node/src/components/deploy_acceptor.rs
Comment (darthsiroftardis) [reply node/src/components/deploy_acceptor.rs:134]: Added comments

File: node/src/components/deploy_acceptor.rs
Comment (darthsiroftardis) [reply node/src/components/deploy_acceptor.rs:86]: Added comments

**Comment by EdHastingsCasperLabs** [comment]
this is going to conflict w/ https://github.com/CasperLabs/casper-node/pull/694

**Review by EdHastingsCasperLabs — APPROVED** [review]
It looks like the comments you added got removed at some point; approved assuming you add them back before merging.

**Pull request merged**
